src/main/java/com/spotify/reaper/service/RepairRunner.java

```diff
@@ -56,10 +56,18 @@ public static void resumeRunningRepairRuns(IStorage storage,
       JmxConnectionFactory jmxConnectionFactory) {
     for (RepairRun repairRun : storage.getAllRunningRepairRuns()) {
       while (true) {
-        RepairSegment runningSegment = storage.getTheRunningSegment(repairRun.getId());
-        if (runningSegment == null) {
+        Collection<RepairSegment> runningSegments =
+            storage.getSegmentsWithStateForRun(repairRun.getId(), RepairSegment.State.RUNNING);
+        if (runningSegments.size() < 1) {
+          break;
+        }
+        if (runningSegments.size() > 1) {
+          LOG.error("there is more than one running segment on run with id {}, which is not "
```

Comment:
Regardless, we should abort/postpone all the segments that have state RUNNING. That's why this was (and is) a while loop. With the getSegmentsWithStateForRun method, we can turn this into a nicer foreach loop!